stellargraph/datasets/dataset_loader.py

```diff
@@ -49,6 +49,7 @@ class DatasetLoader:
     description = ""
     source = ""
     data_subdirectory_name: Optional[str] = None
+    url_contains_directory: bool = True
```

Comment:
Maybe this could be specified just after the other `url_...` parameters, just to keep the "idea" close together (and the same move in `IAEnronEmployees`)? I think everything uses keyword arguments here, so shuffling/shifting the existing arguments shouldn't break any other code)

Also, since it only refers to archives, maybe it could be:

```suggestion
    url_archive_contains_directory: bool = True
```

It could potentially be `url_archive_contains_base_directory` or `url_archive_contains_top_level_directory` too, but I don't think that matters so much.

Reply:
The parameters that precede it don't have default values though, whereas I think this one probably makes sense to have a default value of `True`, so it can't be moved in front of them

Reply:
I can still move where we refer to it in `IAEnronEmployees` class to keep the `url` related things together though, since all parameters are referred to via keyword arguments

Reply:
Ah, interesting. We could potentially make all the arguments keyword-only, which would allow one with a default to be before one without, e.g.: `def f(x, *, y=1, z): pass` is accepted. That is,

```python
 def __init_subclass__(
    self,
    *,
    name: str,
    ...
```

Reply:
okay I can do that here 👍